server/src/main/java/org/elasticsearch/cluster/routing/allocation/Explanations.java

```diff
@@ -0,0 +1,106 @@
+
+        public static final String YES = """
+            Elasticsearch can allocate the shard.""";
+
+        public static final String THROTTLED = """
+            Elasticsearch is currently busy with other activities, but expects to be able to allocate this shard when those activities \
+            finish. Please wait.""";
+
+        public static final String AWAITING_INFO = """
+            Elasticsearch is currently retrieving information about this shard from one or more nodes, and will make an allocation \
+            decision once it receives the missing information. Please wait.""";
+
+        public static final String ALL_COPIES_INVALID = """
+            Elasticsearch cannot allocate this shard because all the copies it found in the cluster are stale or corrupt. Elasticsearch \
+            will allocate this shard when a node holding a good copy of its data joins the cluster. If no such node is available then you \
+            should restore this index from a recent snapshot.""";
+
+        public static final String NO_COPIES = """
+            Elasticsearch cannot allocate this shard because it could not find any copies of its data in the cluster. Elasticsearch \
+            will allocate this shard when a node holding a good copy of its data joins the cluster. If no such node is available then you \
+            should restore this index from a recent snapshot.""";
+
+        public static final String DELAYED_WITH_ALTERNATIVE = """
+            The node holding this shard copy left the cluster and Elasticsearch is waiting for it to return. If the node does not return \
+            within [%s] then Elasticsearch will be able to allocate this shard to another node in the cluster.""";
+
+        public static final String DELAYED_WITHOUT_ALTERNATIVE = """
+            The node holding this shard copy left the cluster and Elasticsearch is waiting for it to return. If the node does not return \
+            within [%s] then Elasticsearch will not be able to allocate this shard to another node in the cluster.""";
+
+        public static final String EXISTING_STORES_FORBIDDEN = """
+            Elasticsearch is not permitted to allocate this shard to any of the nodes in the cluster that hold an in-sync copy of its \
+            data. If you expect this shard to be allocated to one of the nodes in the cluster then you should find this node in the \
+            node-by-node explanation and address the reasons which prevent Elasticsearch from allocating this shard there.""";
+
+        public static final String ALL_NODES_FORBIDDEN = """
+            Elasticsearch is not permitted to allocate this shard to any of the nodes in the cluster. If you expect this shard to be \
+            allocated to one of the nodes in the cluster then you should find this node in the node-by-node explanation and address the \
+            reasons which prevent Elasticsearch from allocating this shard there.""";
```

Comment:
The error message seems to reference a specific node, but I don't think there is one in this case.
```suggestion
            Elasticsearch isn't allowed to allocate this shard to any node in the cluster. If you expect the shard to be allocated, \
            review the node-by-node explanation and address the reasons that prevent Elasticsearch from allocating the shard.""";
```

Reply:
This is something that I find users struggle with: you have to pick a specific node to which you think the shard should be allocated and ask "why is the shard not on this node". Otherwise it's too hard to make sense of all the information returned by this API. I clarified this further in the places that aren't conditional: `Choose a node...`